src/js/Calculator.js

```diff
@@ -0,0 +1,103 @@
+import { BUTTON_TYPE, ERROR_MESSAGES, OPERATOR, VALUE } from "./constants.js";
+
+const $ = (target) => {
+  return document.querySelector(target);
+};
+
+class Calculator {
+  constructor() {
+    this.curNumber = "";
+    this.prevNumber = "";
+    this.operator = "";
+    this.init();
+  }
+
+  init() {
+    this.registerButtonClickHandler();
+  }
+
+  registerButtonClickHandler = () => {
+    const buttons = document.querySelectorAll("button");
+    const { DIGIT, OPERATION, MODIFIER } = BUTTON_TYPE;
+
+    buttons.forEach((button) => {
+      if (button.classList.contains(DIGIT)) {
+        button.addEventListener("click", () => this._onClickDigit(button));
+      }
+      if (button.classList.contains(OPERATION)) {
+        button.addEventListener("click", () => this._onClickOperator(button));
+      }
+      if (button.classList.contains(MODIFIER)) {
+        button.addEventListener("click", this._onClickModifier);
+      }
+    });
+  };
+
+  _render() {
+    if (this.prevNumber) {
+      $("#total").innerHTML = `${this.prevNumber}${this.operator}${this.curNumber}`;
+      return;
+    }
+    if (this.curNumber) {
+      $("#total").innerHTML = `${this.curNumber}${this.operator}`;
+      return;
+    }
+    $("#total").innerHTML = VALUE.DEFAULT_VALUE;
+  }
+
+  _onClickDigit = (target) => {
+    if (this.curNumber.length >= VALUE.DIGIT_LIMIT_LENGTH) {
+      return alert(ERROR_MESSAGES.DIGIT_LIMIT);
+    }
+    this.curNumber += target.innerHTML;
+    this._render();
+  };
+
+  _onClickOperator = (target) => {
+    if (target.innerHTML === OPERATOR.EQUAL) {
+      const total = this._calculate(this.operator);
+      this.curNumber = parseInt(total);
+      this.prevNumber = "";
+      this.operator = "";
+      this._render();
+      return;
+    }
+    this.prevNumber = this.curNumber;
+    this.curNumber = "";
+    this.operator = target.innerHTML;
+    this._render();
+  };
+
+  _onClickModifier = () => {
+    this.curNumber = "";
+    this.operator = "";
+    this.prevNumber = "";
+    this._render();
+  };
+
+  _calculate = (operator) => {
+    const curNumber = this.curNumber ? parseInt(this.curNumber) : VALUE.DEFAULT_VALUE;
+    const prevNumber = parseInt(this.prevNumber);
```

Comment:
```suggestion
    const prevNumber = +this.prevNumber;
```
문자열을 인트형으로 바꿀 때 제가 주로 사용하는 방법입니다. `parseInt` 와는 어떤 차이가 있는지는 솔직히 잘 모르겠습니다만, 간편하게 사용하기에 좋아보여요

Reply:
간편한 방법이 있었군요 감사합니다 !